examples/context-based-restrictions.v1.test.js

```diff
@@ -0,0 +1,528 @@
+    // begin-replace_zone
+
+    // Request models needed by this operation.
+
+    // AddressIPAddress
+    const addressModel = {
+      type: 'ipAddress',
+      value: '169.23.57.235',
+    };
+
+    const params = {
+      zoneId,
+      ifMatch: zoneEtag,
+      name: 'an example of zone to be replaced',
+      accountId,
+      addresses: [addressModel],
+      description: 'this is an example of zone to be replaced',
+    };
+
+    await contextBasedRestrictionsService
+      .replaceZone(params)
+      .then((res) => {
+        zoneId = res.result.id;
+        zoneEtag = res.headers.etag;
+
+        console.log(JSON.stringify(res.result, null, 2));
+      })
+      .catch((err) => {
+        console.warn(err);
+      });
+
+    // end-replace_zone
+  });
+
+  test('listAvailableServicerefTargets request example', async (done) => {
+    consoleLogMock.mockImplementation((output) => {
+      originalLog(output);
+      done();
+    });
+    consoleWarnMock.mockImplementation((output) => {
+      done(output);
+    });
+
+    originalLog('listAvailableServicerefTargets() result:');
+    // begin-list_available_serviceref_targets
+
+    await contextBasedRestrictionsService
+      .listAvailableServicerefTargets({})
+      .then((res) => {
+        console.log(JSON.stringify(res.result, null, 2));
+      })
+      .catch((err) => {
+        console.warn(err);
+      });
+
+    // end-list_available_serviceref_targets
+  });
+
+  test('createRule request example', async (done) => {
+    consoleLogMock.mockImplementation((output) => {
+      originalLog(output);
+      done();
+    });
+    consoleWarnMock.mockImplementation((output) => {
+      done(output);
+    });
+
+    originalLog('createRule() result:');
+    // begin-create_rule
+
+    // Request models needed by this operation.
+
+    // RuleContextAttribute
+    const ruleContextAttributeModel = {
+      name: 'networkZoneId',
+      value: zoneId,
+    };
+
+    // RuleContext
+    const ruleContextModel = {
+      attributes: [ruleContextAttributeModel],
+    };
+
+    // ResourceAttribute
+    const resourceAttributeModel = {
+      name: 'accountId',
+      value: accountId,
+    };
+
+    const resourceAttributeServiceNameModel = {
+      name: 'serviceName',
+      value: serviceName,
+      operator: 'stringEquals',
+    };
+
+    // Resource
+    const resourceModel = {
+      attributes: [resourceAttributeModel, resourceAttributeServiceNameModel],
+    };
+
+    const params = {
+      contexts: [ruleContextModel],
+      resources: [resourceModel],
+      description: 'this is an example of rule',
+    };
+
+    await contextBasedRestrictionsService
+      .createRule(params)
+      .then((res) => {
+        ruleId = res.result.id;
+        ruleEtag = res.headers.etag;
+
+        console.log(JSON.stringify(res.result, null, 2));
+      })
+      .catch((err) => {
+        console.warn(err);
+      });
+
+    // end-create_rule
+  });
```

Comment:
I'd like to ask you to remove the comments between `// end-create_rule` and `// begin-create_rule` lines. We don't want to display comments in the reference.

Reply:
removed